bundles/net.tourbook/src/net/tourbook/ui/views/nutrition/DialogSearchProduct.java

```diff
@@ -0,0 +1,861 @@
+             * Button: Add
+             */
+            _btnAdd = new Button(headerContainer, SWT.PUSH);
+            _btnAdd.setText(Messages.Dialog_SearchProduct_Button_Add);
+            _btnAdd.setToolTipText(Messages.Dialog_SearchProduct_Button_Add_Tooltip);
+            _btnAdd.addSelectionListener(widgetSelectedAdapter(selectionEvent -> onAddProduct()));
+            setButtonLayoutData(_btnAdd);
+         }
+         {
+            /*
+             * Label: Search Type
+             */
+            _lblSearchType = UI.createLabel(headerContainer, Messages.Dialog_SearchProduct_Label_SearchType);
+            GridDataFactory.fillDefaults()
+                  .align(SWT.LEFT, SWT.CENTER)
+                  .applyTo(_lblSearchType);
+         }
+         {
+            /*
+             * combo: Search type
+             */
+            _comboSearchType = new Combo(headerContainer, SWT.READ_ONLY);
+            _comboSearchType.setVisibleItemCount(2);
+            _comboSearchType.addSelectionListener(widgetSelectedAdapter(selectionEvent -> validateFields()));
+            GridDataFactory.fillDefaults()
+                  .align(SWT.LEFT, SWT.CENTER)
+                  .grab(true, false)
+                  .applyTo(_comboSearchType);
+         }
+      }
+
+      _queryViewer = new ComboViewer(_comboSearchQuery);
+      _queryViewer.setContentProvider(new SearchContentProvider());
+      _queryViewer.setComparator(new ViewerComparator());
+   }
+
+   private void createUI_20_Viewer(final Composite parent) {
+
+      /*
+       * table viewer: products
+       */
+      final Table productsTable = new Table(parent, SWT.SINGLE | SWT.FULL_SELECTION);
+      GridDataFactory.fillDefaults().grab(true, true).applyTo(productsTable);
+      productsTable.setLinesVisible(_prefStore.getBoolean(ITourbookPreferences.VIEW_LAYOUT_DISPLAY_LINES));
+      productsTable.setHeaderVisible(true);
+
+      final GridData gdDescription = (GridData) productsTable.getLayoutData();
+      gdDescription.heightHint = _pc.convertHeightInCharsToPixels(15);
+
+      // column: Barcode
+      final TableColumn columnCode = new TableColumn(productsTable, SWT.TRAIL);
+      columnCode.setText(Messages.Tour_Nutrition_Column_Code);
+      columnCode.setWidth(_pc.convertWidthInCharsToPixels(20));
+      columnCode.addListener(SWT.Selection, _columnSortListener);
+      // column: Name
+      final TableColumn columnName = new TableColumn(productsTable, SWT.TRAIL);
+      columnName.setText(Messages.Tour_Nutrition_Column_Name);
+      columnName.setWidth(_pc.convertWidthInCharsToPixels(55));
+      columnName.addListener(SWT.Selection, _columnSortListener);
+      // column: Quantity
+      final TableColumn columnQuantity = new TableColumn(productsTable, SWT.TRAIL);
+      columnQuantity.setText(Messages.Tour_Nutrition_Column_Quantity);
+      columnQuantity.setWidth(_pc.convertWidthInCharsToPixels(15));
+      columnQuantity.addListener(SWT.Selection, _columnSortListener);
```

Comment:
I don't like the bar code column, it's too much noise for me.

Can you implement these columns with the column manager, that the user can customize this view. One of my TODO list is, that the alignment can also be customized but this is not yet available.

![image](https://github.com/mytourbook/mytourbook/assets/1283445/734488f8-39c4-422b-a3a0-99922894d15c)

vs which I would like to display

![image](https://github.com/mytourbook/mytourbook/assets/1283445/5eaf633c-86af-475b-b2fd-72641d8aa8a2)

or

![image](https://github.com/mytourbook/mytourbook/assets/1283445/908f9dde-a934-4873-9f69-3201874dcaf1)

vs

![image](https://github.com/mytourbook/mytourbook/assets/1283445/cb1d7fa6-cd80-4371-866c-f900e806e854)


There is a trick that the first visible column is right aligned because the first column is ALWAYS left aligned. I created an empty 1st column which is not visible.

Reply:
>Can you implement these columns with the column manager, that the user can customize this view

Feature implemented

>One of my TODO list is, that the alignment can also be customized but this is not yet available.

I think this will be a good improvement

Reply:
> > Can you implement these columns with the column manager, that the user can customize this view
> 
> Feature implemented

After customizing the columns, the viewer is not as it should:

* Too much vertical space
* No border between the viewer and the dialog

I think that the parent of the viewer is wrong

This viewer needs a new invisible column with the width = 0 that the first visible column can be right aligned

![image](https://github.com/mytourbook/mytourbook/assets/1283445/fc9ebf8b-4892-402d-a6c1-b2145babf5cd)

Reply:
>This viewer needs a new invisible column with the width = 0 that the first visible column can be right aligned

I did one and while it works and it is a very "hacky" trick, the issue is that when customizing the table, it can be confusing to the user

![image](https://github.com/mytourbook/mytourbook/assets/25235665/5e6ffe4d-b92b-4df3-a18d-11454aa12dd3)

Reply:
>After customizing the columns, the viewer is not as it should:

>    Too much vertical space
>    No border between the viewer and the dialog

> I think that the parent of the viewer is wrong

Those issues should be fixed now

Reply:
> I did one and while it works and it is a very "hacky" trick, the issue is that when customizing the table, it can be confusing to the user

The "Alignment" column should 

* not be the first column, by default it should be the last column
* not be a default column, so by default it is hidden
* have a label/header 
* have a width of 0

Then I think for a normal user it is less confusing

Reply:
>not be the first column, by default it should be the last column
When I move the "Alignment" column to be the last column, then the alignment doesn't work anymore

![image](https://github.com/mytourbook/mytourbook/assets/25235665/3ed6386c-08a0-41c4-a599-a2f17c5b5d9f)

Reply:
>> not be the first column, by default it should be the last column

> When I move the "Alignment" column to be the last column, then the alignment doesn't work anymore

This is correct but my suggestion was for advanced users who understand why this column is needed

With the alignment column I can now customize this viewer and I would be happy when the customization is also saved and restored.

![image](https://github.com/mytourbook/mytourbook/assets/1283445/9ad32c79-4c5f-4f11-a3dd-2d0dfda4df01)


The alignment column should have a label similar like

![image](https://github.com/mytourbook/mytourbook/assets/1283445/7c0518fe-0ad2-402d-a0e1-0976702aed2b)

Reply:
>With the alignment column I can now customize this viewer and I would be happy when the customization is also saved and restored.

This is fixed now

>The alignment column should have a label similar like

Sure, should it be internationalized ?

Reply:
> > The alignment column should have a label similar like
> 
> Sure, should it be internationalized ?

Yes, it's a almost a "normal" column

Reply:
Fixed